apollo-ast/src/main/kotlin/com/apollographql/apollo3/ast/check_key_fields.kt

```diff
@@ -33,7 +33,9 @@ private fun CheckKeyFieldsScope.checkField(
     selections: List<GQLSelection>,
     parentType: String,
 ) {
-  schema.typeDefinitions.values.filterIsInstance<GQLObjectTypeDefinition>().forEach {
+  schema.typeDefinitions.values.filter {
+    it is GQLObjectTypeDefinition || it is GQLInterfaceTypeDefinition
+  }.forEach {
```

Comment:
I don't think we need to iterate the interfaces here, this check is used to detect cases like this:

```graphql
# schema.graphqls
extend type Cat @typePolicy(keyFields: "id")
```

```graphql
# operation.graphql
query GetCat {
  animal {
    # id is missing here
    species
  }
}
```

So it only cares about actual concrete objects because at the end of the day, all objects must have a concrete type.